src/NServiceBus.Persistence.CosmosDB.LogicalOutbox.AcceptanceTests/ConfigureEndpointCosmosDBPersistence.cs

```diff
@@ -47,15 +47,5 @@ public override Task Invoke(IIncomingLogicalMessageContext context, Func<Task> n
             context.Extensions.Set(new ContainerInformation(SetupFixture.ContainerName, new PartitionKeyPath(SetupFixture.PartitionPathKey)));
             return next();
         }
-
-        public class PartitionKeyProviderBehaviorRegisterStep : RegisterStep
-        {
-            public PartitionKeyProviderBehaviorRegisterStep() : base(nameof(PartitionKeyProviderBehavior),
-                typeof(PartitionKeyProviderBehavior),
-                "Populates the partition key",
-                b => new PartitionKeyProviderBehavior(b.Build<ScenarioContext>()))
-            {
```

Comment:
We removed `InsertStepBeforeIfExists` from here. And it works. Is this chance?

We absolutely need this behavior to run before the `LogicalOutboxBehavior`, is this guaranteed in some way because it is a "user" registration?

Reply:
As far as I remember core does a topological sort of all the registrations. I'm not sure though if the registration sorting is a stable sort that always guarantees the same order though